src/GafferArnold/IECoreArnoldPreview/ShaderNetworkAlgo.cpp

```diff
@@ -354,6 +356,14 @@ std::vector<AtNode *> convert( const IECoreScene::ShaderNetwork *shaderNetwork,
 			return AiNode( nodeType, nodeName, parentNode );
 		};
 		convertWalk( shaderNetwork->getOutput(), shaderNetwork, name, nodeCreator, result, converted );
+		if( const IECoreScene::Shader *surface = shaderNetwork->outputShader() )
```

Comment:
I'm fairly sure this test is redundant - we've already checked there is an output, so this should be non-null. Certainly, `convertWalk()` isn't doing the equivalent check, so any problem would be apparent before we got here anyway.

Reply:
Removed the `if` in 99c8d27ab84186119f2a5fc9019ec787401bcabb